Extensions/NavMeshPathfindingBehavior/A_navmeshgenerator.ts

```diff
@@ -0,0 +1,3411 @@
+// This implementation is strongly inspired from a Java one
+// by Stephen A. Pratt:
+// http://www.critterai.org/projects/nmgen_study/
+//
+// Most of the comments were written by him and were adapted to fit this implementation.
+// This implementation differs a bit from the original:
+// - it's only 2D instead of 3D
+// - it has less features (see TODO) and might have lesser performance
+// - it uses objects for points instead of pointer-like in arrays of numbers
+// - the rasterization comes from other sources because of the 2d focus
+// - partialFloodRegion was rewritten to fix an issue
+// - filterNonObstacleVertices was added
+//
+// The Java implementation was also inspired from Recast that can be found here:
+// https://github.com/recastnavigation/recastnavigation
+
+namespace gdjs {
+  /**
+   * A cell that holds data needed by the 1st steps of the NavMesh generation.
+   */
+  export class RasterizationCell {
```

Comment:
I know this means reworking tests and other files, but can we put all these classes into an additional namespace?
```
namespace gdjs { 
namespace NavMesh {
```

Otherwise, things being on `gdjs` directly would confuse users who would think these are general purpose tools. Also should we in the future develop a version 2 of NavMesh, we'll conflict with the existing classes - that will be terrible!

General rule is simple: Put all your extension tools into a namespace, and reduce the number of stuff on `gdjs` directly to strictly the minimum: behaviors, objects, that's it :)